src/lib/kclCommands.ts

```diff
@@ -0,0 +1,111 @@
+import { CommandBarOverwriteWarning } from 'components/CommandBarOverwriteWarning'
+import { Command, CommandArgumentOption } from './commandTypes'
+import { kclManager } from './singletons'
+import { isDesktop } from './isDesktop'
+import { FILE_EXT } from './constants'
+
+interface OnSubmitProps {
+  sampleName: string
+  code: string
+  method: 'overwrite' | 'newFile'
+}
+
+export function kclCommands(
+  onSubmit: (p: OnSubmitProps) => Promise<void>,
+  providedOptions: CommandArgumentOption<string>[]
+): Command[] {
+  return [
+    {
+      name: 'format-code',
+      displayName: 'Format Code',
+      description: 'Nicely formats the KCL code in the editor.',
+      needsReview: false,
+      groupId: 'code',
+      icon: 'code',
+      onSubmit: () => {
+        kclManager.format()
+      },
+    },
+    {
+      name: 'open-kcl-example',
+      displayName: 'Open sample',
+      description: 'Imports an example KCL program into the editor.',
+      needsReview: true,
+      icon: 'code',
+      reviewMessage: ({ argumentsToSubmit }) =>
+        argumentsToSubmit.method === 'newFile'
+          ? 'Create a new file with the example code?'
+          : CommandBarOverwriteWarning({}),
+      groupId: 'code',
+      onSubmit(data) {
+        if (!data?.sample) {
+          return
+        }
+        const sampleCodeUrl = `https://raw.githubusercontent.com/KittyCAD/kcl-samples/main/${data.sample.replace(
+          FILE_EXT,
+          ''
+        )}/${data.sample}`
```

Comment:
Please use [encodeURIComponent()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) when injecting into a URL.

Reply:
A space in the file name is the most obvious problem. But forgetting to escape is often the source of security issues.